src/pharmpy/tools/estmethod/algorithms.py

```diff
@@ -171,31 +240,37 @@ def _create_candidate_model(
     while len(model.estimation_steps) > 0:
         model = remove_estimation_step(model, 0)
 
-    model = add_estimation_step(model, **est_settings)
-    model = add_estimation_step(model, **eval_settings)
+    if only_evaluation:
+        model = add_estimation_step(model, **eval_settings)
+    elif add_eval_after_est:
+        model = add_estimation_step(model, **est_settings)
+        model = add_estimation_step(model, **eval_settings)
+    else:
+        model = add_estimation_step(model, **est_settings)
+
     if solver:
         model = set_ode_solver(model, solver)
     return model
 
 
-def _create_est_settings(method, only_evaluation=False):
+def _create_est_settings(method, parameter_uncertainty_method=None, add_eval_after_est=False):
     est_settings = {
         'method': method,
         'interaction': True,
         'laplace': False,
         'auto': True,
         'keep_every_nth_iter': 10,
-        'parameter_uncertainty_method': None,
+        'maximum_evaluations': 9999,
     }
 
     if method == 'LAPLACE':
         est_settings['method'] = 'FOCE'
         est_settings['laplace'] = True
 
-    if only_evaluation:
-        est_settings['evaluation'] = True
+    if add_eval_after_est:
+        est_settings['parameter_uncertainty_method'] = None
     else:
-        est_settings['maximum_evaluations'] = 9999
+        est_settings['parameter_uncertainty_method'] = parameter_uncertainty_method
```

Comment:
Why has this condition changed? Can't see where `evaluation=True`?

Reply:
See comment https://github.com/pharmpy/pharmpy/pull/2068#discussion_r1333076138